### ISSUE TITLE
Use something similiar to shell-directory-tracker for directory tracking

### ISSUE THREAD

**Issue by brotzeit** [akermu/emacs-libvterm#466]
I still can't get directory tracking to work and I wonder if it would be better to do it like shell-mode with `shell-dirtrack-mode` and `shell-directory-tracker`. What do you think ?

**Comment by Sbozzolo** [comment]
I am not very familiar with `shell-dirtrack-mode` but it seems to work by tracking `cd`, `pushd` and `popd`. It also seems quite fragile, according to the comments in its code this minimal example:
```
emacs -Q -f shell
foo=/tmp
cd $foo
```
will not work. It doesn't even support aliases.

While it doesn't require messing with the prompt, or post-cmd functions, `shell-dirtrack-mode` appears to much more limited compared to what we already have, which, by the way, is exactly what `term.el` implements. This method relies on receiving data from the shell via escape sequences. Once it is correctly setup, it works very well.

In case modifying your prompt is a problem, at the bottom of the README we describe a way to do directory tracking without changing the shell configuration (using `procfs`). 

Finally, I think that you can use `dirtrack-mode` with `vterm`. If your prompt contains the path of the current working directory, you can extract that with a regexp.

In conclusion, I don't see compelling reasons to add support for `shell-dirtrack-mode`, as there are better alternatives. 

What problems are you facing in setting up directory tracking? For `bash` you only have to add the following lines:
```
vterm_prompt_end(){
    vterm_printf "51;A$(whoami)@$(hostname):$(pwd)"
}
PS1=$PS1'\[$(vterm_prompt_end)\]'
```

**Comment by brotzeit** [comment]
I missed vterm_printf. Now it's working. Thanks for taking the time and great work on the package!
